src/tic_tac_toe/setup.clj

```diff
@@ -19,33 +22,38 @@
   (hash-map :current (human/create-human-player "X")
             :opponent (computer/create-computer-player "O" "X")))
 
-(defn- setup-utils
-  []
-  (let [game-serializer (serializer/create-serializer)
-        game-io (io/create-console-io)]
-    (hash-map :reader (reader/create-reader game-serializer)
-              :writer (writer/create-writer game-serializer)
-              :ui (ui/create-ui game-io)
-              :options options)))
+(defn- setup-location
+  [directory filename utils]
+  (assoc utils :dir directory :file filename))
 
 (defn- setup-saved-game
-  [directory filename utils]
-  ( -> (reader/load-game (:reader utils) directory filename
-                         human/create-human-player
-                         computer/create-computer-player)
-       (merge utils)))
+  [utils]
+  (if (reader/save-exists? (:reader utils) (:dir utils) (:file utils))
+    (-> (reader/load-game (:reader utils) (:dir utils) (:file utils)
```

Comment:
You might take a look at what Clojure has to offer in terms of destructuring function arguments.

Reply:
I've [adjusted](https://github.com/saralein/tic-tac-toe_clojure/pull/10/files#diff-4fbcd66ffb99e118519711e232783759R30) this here and throughout.  Thank you for mentioning it.  I believe I saw it before, but it didn't register.  I'm much happier using it.  Some of the args are a little hairier, but the bodies are easier to read.